city/router.py

```diff
@@ -0,0 +1,63 @@
+from sqlalchemy.orm import Session
+from fastapi import APIRouter, Depends, HTTPException
+
+from . import schemas, crud
+from dependencies import get_db
+
+router = APIRouter()
+
+
+@router.get("/cities", response_model=list[schemas.CityDefault])
+def get_all_cities(db: Session = Depends(get_db)):
+    return crud.get_all_cities(db)
+
+
+@router.get("/cities/{city_id}", response_model=schemas.CityDefault)
+def get_city_by_id(city_id: int, db: Session = Depends(get_db)):
+    db_city = crud.get_city_by_id(db, city_id)
+    if not db_city:
+        raise HTTPException(
+            status_code=400,
+            detail=f"City with id={city_id} was not found"
+        )
+    else:
+        return db_city
+
+
+@router.post("/cities", response_model=schemas.CityDefault,
+             status_code=201,
+             response_description="Successful Response. City was created")
+def create_city(new_city: schemas.CityCreate, db: Session = Depends(get_db)):
+    db_city = crud.get_city_by_name(db, new_city.name)

```

Comment:
We do have city_id on the update method, so why you've decided to use search by name?

Reply:
We have field 'name' in table 'city'. This field has attribute unique=True. But if we try to create new city with the same name, server return 'Error: Internal Server Error' and stutus code 500. It is not informational. So, I want to catch this issue and make it more understanding for user and inform that that name already exists. I want to inform where we have a problem